dev/tests/integration/framework/Magento/TestFramework/Db/DymanicTables/CategoryProductIndexTables.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\TestFramework\Db\DymanicTables;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Store\Model\Store;
+
+/**
+ * Class to pre-create category product index tables
+ */
+class CategoryProductIndexTables
+{
+
+    /**
+     * @var string
+     */
+    private $prototype = 'catalog_category_product_index';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection
+    ) {
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * Creates category product index tables
+     */
+    public function createTables(): void
+    {
+        $connection = $this->resourceConnection->getConnection();
+        for ($storeId = 0; $storeId <= 256; $storeId++) {
```

Comment:
What test we fix with this ad-hoc solution?

Reply:
As I remember it's for dev/tests/integration/testsuite/Magento/CatalogImportExport/Model/ProductTest.php. This code is in mainline already.